### PR TITLE
Disable caching of /ui responses

### PR DESCRIPTION
**Pull request by wendigo**
This ensures that any sensitive information is not cached in the browser

Alternative to https://github.com/trinodb/trino/pull/22480

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/server/ui/DisableCacheResponseFilter.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+@Priority(0)
+public class DisableCacheResponseFilter
+        implements ContainerResponseFilter
+{
+    @Override
+    public void filter(ContainerRequestContext requestContext, ContainerResponseContext responseContext)
+            throws IOException
+    {
+        if (!requestContext.getUriInfo().getPath().startsWith("ui/")) {
```
Comment (losipiuk) [inline core/trino-main/src/main/java/io/trino/server/ui/DisableCacheResponseFilter.java:31]: can it be goverened by annotation instead?

Comment (wendigo) [reply core/trino-main/src/main/java/io/trino/server/ui/DisableCacheResponseFilter.java:31]: Yes but using DynamicFeature

Comment (wendigo) [reply core/trino-main/src/main/java/io/trino/server/ui/DisableCacheResponseFilter.java]: PTAL.

**Review by losipiuk — APPROVED** [review]
Looks nice IMO. Probably we can put the logic in Airlift - but I will not insist.

**Pull request merged**